activesupport/lib/active_support/duration/iso8601_serializer.rb

```diff
@@ -40,7 +41,7 @@ def normalize
           end
           # If all parts are negative - let's make a negative duration
           sign = ''
-          if parts.values.all? { |v| v < 0 }
+          if !parts.empty? && parts.values.all? { |v| v < 0 }
```

Comment:
I disagree, I think that leaving this change in reduces cognitive overhead for whomever comes next in refactoring the `#serialize` method, they'll have to deal with this bug in `#normalize`, too. I'm also skeptical of the cost of "git churn" as a metric. 

But, since I am unlikely to be the one to refactor the `#serialize` method, and would really like to get this bugfix in so I can remove the monkeypatch from my app at some point, I'll revert this change.